src/app/public/modules/inline-form/inline-form.component.spec.ts

```diff
@@ -0,0 +1,297 @@
+import {
+  async,
+  ComponentFixture,
+  TestBed
+} from '@angular/core/testing';
+
+import {
+  By
+} from '@angular/platform-browser';
+
+import {
+  expect
+} from '@skyux-sdk/testing';
+
+import {
+  SkyInlineFormFixtureComponent
+} from './fixtures/inline-form.fixture';
+
+import {
+  SkyInlineFormModule
+} from './inline-form.module';
+
+import {
+  SkyInlineFormType
+} from './types';
+
+import {
+  SkyInlineFormAdapterService
+} from './inline-form-adapter.service';
+
+function getPrimaryButton(fixture: ComponentFixture<any>) {
+  return fixture.debugElement.query(
+    By.css('.sky-inline-form-footer .sky-btn-primary')
+  );
+}
+
+function getDefaultButton(fixture: ComponentFixture<any>) {
+  return fixture.debugElement.query(
+    By.css('.sky-inline-form-footer .sky-btn-default')
+  );
+}
+
+function getLinkButton(fixture: ComponentFixture<any>) {
+  return fixture.debugElement.query(
+    By.css('.sky-inline-form-footer .sky-btn-link')
+  );
+}
+
+function verifyDoneButtonisDefined(fixture: ComponentFixture<any>, isDefined: boolean) {
+  const doneButton = getPrimaryButton(fixture);
+  if (isDefined) {
+    expect(doneButton).not.toBeNull();
+    expect(doneButton.nativeElement.textContent).toContain('Done');
+  } else {
+    if (doneButton) {
+      expect(doneButton.nativeElement.textContent).not.toContain('Done');
+    }
+  }
+}
+
+function verifySaveButtonisDefined(fixture: ComponentFixture<any>, isDefined: boolean) {
+  const saveButton = getPrimaryButton(fixture);
+  if (isDefined) {
+    expect(saveButton).not.toBeNull();
+    expect(saveButton.nativeElement.textContent).toContain('Save');
+  } else {
+    if (saveButton) {
+      expect(saveButton.nativeElement.textContent).not.toContain('Save');
+    }
+  }
+}
+
+function verifyDeleteButtonIsDefined(fixture: ComponentFixture<SkyInlineFormFixtureComponent>, isDefined: boolean) {
+  const deleteButton = getDefaultButton(fixture);
+  if (isDefined) {
+    expect(deleteButton).not.toBeNull();
+    expect(deleteButton.nativeElement.textContent).toContain('Delete');
+  } else {
+    expect(deleteButton).toBeNull();
+  }
+}
+
+function verifyCancelButtonIsDefined(fixture: ComponentFixture<SkyInlineFormFixtureComponent>, isDefined: boolean) {
+  const cancelButton = getLinkButton(fixture);
+  if (isDefined) {
+    expect(cancelButton).not.toBeNull();
+    expect(cancelButton.nativeElement.textContent).toContain('Cancel');
+  } else {
+    expect(cancelButton).toBeNull();
+  }
+}
+
+describe('Inline form component', () => {
+  let component: SkyInlineFormFixtureComponent,
+  fixture: ComponentFixture<SkyInlineFormFixtureComponent>;
+
+  function showForm() {
+    component.showForm = true;
+    fixture.detectChanges();
+  }
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      declarations: [
+        SkyInlineFormFixtureComponent
```

Comment:
It would clean up the `TestBed` configurations if we had a `SkyInlineFormFixtureModule` that you could then import into the `TestBed`.
```
TestBed.configureTestingModule({
  imports: [
    SkyInlineFormFixtureModule
  ]
});
```